descriptor.go

```diff
@@ -0,0 +1,54 @@
+package gogh
+
+import (
+	"context"
+	"errors"
+	"strings"
+)
+
+const DefaultHost = "github.com"
+
+var (
+	ErrTooManySlashes = errors.New("too many slashes")
+)
+
+// Descriptor will parse any string as a Description.
+//
+// If it is clear that the string has host, user and name explicitly,
+// use "NewDescription" instead to build Description.
+type Descriptor struct {
```

Comment:
Descriptor should multi host, multi user.
Because the Parse is the duty of the Descriptor.